gdk/x11/gdkglcontext-x11-eglx.c

```diff
@@ -810,13 +843,28 @@ gdk_x11_display_make_gl_context_current (GdkDisplay   *display,
 
   if (context_x11->is_attached)
     {
-      /* If the WM is compositing there is no particular need to delay
-       * the swap when drawing on the offscreen, rendering to the screen
-       * happens later anyway, and its up to the compositor to sync that
-       * to the vblank. */
       GdkScreen *screen = gdk_window_get_screen (window);
+      gboolean is_composited = gdk_screen_is_composited (screen);
+
+#ifdef HAVE_XCOMPOSITE
```

Comment:
The `HAVE_XCOMPOSITE` define is for using the XComposite API, not for the existence of the XComposite extension on the server.